### PR TITLE
Unwrap errors

### PR DESCRIPTION
**Pull request by olegsu**
Related to #388 

This PR includes also:
1. Update all the logger calls where the regular `go vet` would report
2. Update the CI script `golangci-lint` configuration to run custom `print.funcs`

### Why update vet with print.funcs?
Normally `go vet` runs on a lot of functions signatures (the full list `go tool vet help printf`), but not the one that in [logp](https://github.com/elastic/elastic-agent-libs/blob/main/logp/logger.go)
To overcome this `go vet` allow to add additional functions to check.
By setting `-printfuncs="Infof,Debugf,Warnf,Errorf,Fatalf,Panicf,DPanicf"` we can make sure that all the logger calls are covered

### Run locally
* Install [golangci-lint ](https://golangci-lint.run/usage/install/#macos)
* Run `golangci-lint run`

### Output
When having formatting issues, for example `l.log.Infof("Waiting for initial reconfiguration from Fleet server... %s")`
You would see
image

### Additional Reading
* [go vet cmd](https://pkg.go.dev/cmd/vet)
* [wrap-unwrap-errors](https://gosamples.dev/wrap-unwrap-errors/)
* [golang ci - go vet](https://golangci-lint.run/usage/linters/#govet)

### CODE REVIEW

File: .github/workflows/cloudbeat-ci.yml [hunk 1]
```diff
@@ -230,6 +230,7 @@ jobs:
       - name: Run Tests
         id: run_tests
         run: |
+          just run-golang-vet
```
[the remark this answers is not in the document]
Comment (oren-zohar) [reply .github/workflows/cloudbeat-ci.yml:233]: should probably be part of `unit-test.yml` so we'll run it only once

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: .github/workflows/cloudbeat-ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (olegsu) [reply .github/workflows/cloudbeat-ci.yml:233]: I see we are using golangci-lint. Will update its config

File: .golangci.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      printf:
+        funcs:
+          - Infof
+          - Debugf
+          - Warnf
+          - Errorf
+          - Fatalf
+          - Panicf
+          - DPanicf
+    disable-all: true
```
Comment (oren-zohar) [inline .golangci.yaml:17]: Why not include them if we are passing all of the analyzers? (Tested locally and appears to pass)

**Review by oren-zohar — CHANGES_REQUESTED** [review]
looks great, can you please share a screenshot of what the logs looks like for one of the edited logs?

File: .golangci.yaml (hunk 2 above)
Comment (olegsu) [reply .golangci.yaml:17]: Missed this one

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**